test/lit/passes/global-effects.wast

```diff
@@ -293,84 +286,71 @@
           (ref.func $throw)
         )
       )
-      (catch_all)
     )
   )
 
   ;; WITHOUT:      (func $call-return-call-throw-and-catch (type $void)
-  ;; WITHOUT-NEXT:  (try
-  ;; WITHOUT-NEXT:   (do
+  ;; WITHOUT-NEXT:  (block $tryend
+  ;; WITHOUT-NEXT:   (try_table (catch_all $tryend)
   ;; WITHOUT-NEXT:    (call $return-call-throw-and-catch)
   ;; WITHOUT-NEXT:   )
-  ;; WITHOUT-NEXT:   (catch_all
-  ;; WITHOUT-NEXT:    (nop)
-  ;; WITHOUT-NEXT:   )
   ;; WITHOUT-NEXT:  )
-  ;; WITHOUT-NEXT:  (try
-  ;; WITHOUT-NEXT:   (do
+  ;; WITHOUT-NEXT:  (block $tryend0
+  ;; WITHOUT-NEXT:   (try_table (catch_all $tryend0)
   ;; WITHOUT-NEXT:    (call $return-call-indirect-throw-and-catch)
   ;; WITHOUT-NEXT:   )
-  ;; WITHOUT-NEXT:   (catch_all
-  ;; WITHOUT-NEXT:    (nop)
-  ;; WITHOUT-NEXT:   )
   ;; WITHOUT-NEXT:  )
-  ;; WITHOUT-NEXT:  (try
-  ;; WITHOUT-NEXT:   (do
+  ;; WITHOUT-NEXT:  (block $tryend1
+  ;; WITHOUT-NEXT:   (try_table (catch_all $tryend1)
   ;; WITHOUT-NEXT:    (call $return-call-ref-throw-and-catch)
   ;; WITHOUT-NEXT:   )
-  ;; WITHOUT-NEXT:   (catch_all
-  ;; WITHOUT-NEXT:    (nop)
-  ;; WITHOUT-NEXT:   )
   ;; WITHOUT-NEXT:  )
   ;; WITHOUT-NEXT:  (call $return-call-throw-and-catch)
   ;; WITHOUT-NEXT:  (call $return-call-indirect-throw-and-catch)
   ;; WITHOUT-NEXT:  (call $return-call-ref-throw-and-catch)
   ;; WITHOUT-NEXT: )
   ;; INCLUDE:      (func $call-return-call-throw-and-catch (type $void)
-  ;; INCLUDE-NEXT:  (try
-  ;; INCLUDE-NEXT:   (do
-  ;; INCLUDE-NEXT:    (call $return-call-indirect-throw-and-catch)
+  ;; INCLUDE-NEXT:  (block $tryend
+  ;; INCLUDE-NEXT:   (try_table (catch_all $tryend)
+  ;; INCLUDE-NEXT:    (call $return-call-throw-and-catch)
   ;; INCLUDE-NEXT:   )
-  ;; INCLUDE-NEXT:   (catch_all
-  ;; INCLUDE-NEXT:    (nop)
+  ;; INCLUDE-NEXT:  )
+  ;; INCLUDE-NEXT:  (block $tryend0
+  ;; INCLUDE-NEXT:   (try_table (catch_all $tryend0)
+  ;; INCLUDE-NEXT:    (call $return-call-indirect-throw-and-catch)
   ;; INCLUDE-NEXT:   )
   ;; INCLUDE-NEXT:  )
-  ;; INCLUDE-NEXT:  (try
-  ;; INCLUDE-NEXT:   (do
+  ;; INCLUDE-NEXT:  (block $tryend1
+  ;; INCLUDE-NEXT:   (try_table (catch_all $tryend1)
   ;; INCLUDE-NEXT:    (call $return-call-ref-throw-and-catch)
   ;; INCLUDE-NEXT:   )
-  ;; INCLUDE-NEXT:   (catch_all
-  ;; INCLUDE-NEXT:    (nop)
-  ;; INCLUDE-NEXT:   )
   ;; INCLUDE-NEXT:  )
   ;; INCLUDE-NEXT:  (call $return-call-throw-and-catch)
   ;; INCLUDE-NEXT:  (call $return-call-indirect-throw-and-catch)
   ;; INCLUDE-NEXT:  (call $return-call-ref-throw-and-catch)
   ;; INCLUDE-NEXT: )
   (func $call-return-call-throw-and-catch
-    (try
-      (do
+    (block $tryend
+      (try_table (catch_all $tryend)
         ;; Even though the body of the previous function is a try-catch_all, the
```

Comment:
```suggestion
        ;; Even though the body of the previous function has a catch_all, the
```
By the way I'm not sure what the "previous function" here (and the original test) refers to. @kripken